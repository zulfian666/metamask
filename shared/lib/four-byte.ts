import { MethodRegistry } from 'eth-method-registry';
import fetchWithCache from './fetch-with-cache';

type FourByteResult = {
  created_at: string;
  text_signature: string;
};

type FourByteResponse = {
  results: FourByteResult[];
};

export async function getMethodFrom4Byte(
  fourBytePrefix: string,
): Promise<string> {
  const fourByteResponse = (await fetchWithCache({
    url: `https://www.4byte.directory/api/v1/signatures/?hex_signature=${fourBytePrefix}`,
    fetchOptions: {
      referrerPolicy: 'no-referrer-when-downgrade',
      body: null,
      method: 'GET',
      mode: 'cors',
    },
    functionName: 'getMethodFrom4Byte',
  })) as FourByteResponse;

  fourByteResponse.results.sort((a, b) => {
    return new Date(a.created_at).getTime() < new Date(b.created_at).getTime()
      ? -1
      : 1;
  });

  return fourByteResponse.results[0].text_signature;
}

let registry: MethodRegistry | undefined;

type HttpProvider = {
  host: string;
  timeout: number;
};

type MethodRegistryArgs = {
  network: string;
  provider: HttpProvider;
};

export async function getMethodDataAsync(
  fourBytePrefix: string,
  allow4ByteRequests: boolean,
  provider?: unknown,
) {
  try {
    let fourByteSig = null;
    if (allow4ByteRequests) {
      fourByteSig = await getMethodFrom4Byte(fourBytePrefix).catch((e) => {
        console.error(e);
        return null;
      });
      console.log('fourByteSig = ', fourByteSig);
    }

    if (!registry) {
      registry = new MethodRegistry({
        provider: provider ?? global.ethereumProvider,
      } as MethodRegistryArgs);
    }

    if (!fourByteSig) {
      return {};
    }

    const parsedResult = registry.parse(fourByteSig);

    return {
      name: parsedResult.name,
      params: parsedResult.args,
    };
  } catch (error) {
    console.error(error);
    return {};
  }
}
