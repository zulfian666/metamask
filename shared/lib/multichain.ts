import { isEthAddress } from '../../app/scripts/lib/multichain/address';

/**
 * Returns whether an address is on the Bitcoin mainnet.
 *
 * This function only checks the prefix of the address to determine if it's on
 * the mainnet or not. It doesn't validate the address itself, and should only
 * be used as a temporary solution until this information is included in the
 * account object.
 *
 * @param address - The address to check.
 * @returns `true` if the address is on the Bitcoin mainnet, `false` otherwise.
 */
export function isBtcMainnetAddress(address: string): boolean {
  return (
    !isEthAddress(address) &&
    (address.startsWith('bc1') || address.startsWith('1'))
  );
}

/**
 * Returns whether an address is on the Bitcoin testnet.
 *
 * See {@link isBtcMainnetAddress} for implementation details.
 *
 * @param address - The address to check.
 * @returns `true` if the address is on the Bitcoin testnet, `false` otherwise.
 */
export function isBtcTestnetAddress(address: string): boolean {
  return !isEthAddress(address) && !isBtcMainnetAddress(address);
}
