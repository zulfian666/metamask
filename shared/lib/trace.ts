import * as Sentry from '@sentry/browser';
import { Primitive, StartSpanOptions } from '@sentry/types';
import { createModuleLogger } from '@metamask/utils';
import { log as sentryLogger } from '../../app/scripts/lib/setupSentry';

export enum TraceName {
  BackgroundConnect = 'Background Connect',
  DeveloperTest = 'Developer Test',
  FirstRender = 'First Render',
  GetState = 'Get State',
  InitialActions = 'Initial Actions',
  LoadScripts = 'Load Scripts',
  Middleware = 'Middleware',
  NestedTest1 = 'Nested Test 1',
  NestedTest2 = 'Nested Test 2',
  NotificationDisplay = 'Notification Display',
  PPOMValidation = 'PPOM Validation',
  SetupStore = 'Setup Store',
  Transaction = 'Transaction',
  UIStartup = 'UI Startup',
}

const log = createModuleLogger(sentryLogger, 'trace');

const ID_DEFAULT = 'default';
const OP_DEFAULT = 'custom';

const tracesByKey: Map<string, PendingTrace> = new Map();

type PendingTrace = {
  end: (timestamp?: number) => void;
  request: TraceRequest;
  startTime: number;
};

export type TraceContext = unknown;

export type TraceCallback<T> = (context?: TraceContext) => T;

export type TraceRequest = {
  data?: Record<string, number | string | boolean>;
  id?: string;
  name: TraceName;
  parentContext?: TraceContext;
  startTime?: number;
  tags?: Record<string, number | string | boolean>;
};

export type EndTraceRequest = {
  id?: string;
  name: TraceName;
  timestamp?: number;
};

export function trace<T>(request: TraceRequest, fn: TraceCallback<T>): T;

export function trace(request: TraceRequest): TraceContext;

export function trace<T>(
  request: TraceRequest,
  fn?: TraceCallback<T>,
): T | TraceContext {
  if (!fn) {
    return startTrace(request);
  }

  return traceCallback(request, fn);
}

export function endTrace(request: EndTraceRequest) {
  const { name, timestamp } = request;
  const id = getTraceId(request);
  const key = getTraceKey(request);
  const pendingTrace = tracesByKey.get(key);

  if (!pendingTrace) {
    log('No pending trace found', name, id);
    return;
  }

  pendingTrace.end(timestamp);

  tracesByKey.delete(key);

  const { request: pendingRequest, startTime } = pendingTrace;
  const endTime = timestamp ?? getPerformanceTimestamp();
  const duration = endTime - startTime;

  log('Finished trace', name, id, duration, { request: pendingRequest });
}

function traceCallback<T>(request: TraceRequest, fn: TraceCallback<T>): T {
  const { name } = request;

  const callback = (span: Sentry.Span | null) => {
    log('Starting trace', name, request);

    const start = Date.now();
    let error: unknown;

    return tryCatchMaybePromise<T>(
      () => fn(span),
      (currentError) => {
        error = currentError;
        throw currentError;
      },
      () => {
        const end = Date.now();
        const duration = end - start;

        log('Finished trace', name, duration, { error, request });
      },
    ) as T;
  };

  return startSpan(request, (spanOptions) =>
    Sentry.startSpan(spanOptions, callback),
  );
}

function startTrace(request: TraceRequest): TraceContext {
  const { name, startTime: requestStartTime } = request;
  const startTime = requestStartTime ?? getPerformanceTimestamp();
  const id = getTraceId(request);

  const callback = (span: Sentry.Span | null) => {
    const end = (timestamp?: number) => {
      span?.end(timestamp);
    };

    const pendingTrace = { end, request, startTime };
    const key = getTraceKey(request);
    tracesByKey.set(key, pendingTrace);

    log('Started trace', name, id, request);

    return span;
  };

  return startSpan(request, (spanOptions) =>
    Sentry.startSpanManual(spanOptions, callback),
  );
}

function startSpan<T>(
  request: TraceRequest,
  callback: (spanOptions: StartSpanOptions) => T,
) {
  const { data: attributes, name, parentContext, startTime, tags } = request;
  const parentSpan = (parentContext ?? null) as Sentry.Span | null;

  const spanOptions: StartSpanOptions = {
    attributes,
    name,
    op: OP_DEFAULT,
    parentSpan,
    startTime,
  };

  return Sentry.withIsolationScope((scope) => {
    scope.setTags(tags as Record<string, Primitive>);

    return callback(spanOptions);
  });
}

function getTraceId(request: TraceRequest) {
  return request.id ?? ID_DEFAULT;
}

function getTraceKey(request: TraceRequest) {
  const { name } = request;
  const id = getTraceId(request);

  return [name, id].join(':');
}

function getPerformanceTimestamp(): number {
  return performance.timeOrigin + performance.now();
}

function tryCatchMaybePromise<T>(
  tryFn: () => T,
  catchFn: (error: unknown) => void,
  finallyFn: () => void,
): T | undefined {
  let isPromise = false;

  try {
    const result = tryFn() as T;

    if (result instanceof Promise) {
      isPromise = true;
      return result.catch(catchFn).finally(finallyFn) as T;
    }

    return result;
  } catch (error) {
    if (!isPromise) {
      catchFn(error);
    }
  } finally {
    if (!isPromise) {
      finallyFn();
    }
  }

  return undefined;
}
