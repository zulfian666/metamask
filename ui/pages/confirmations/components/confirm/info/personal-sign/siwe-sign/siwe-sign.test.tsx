import React from 'react';
import configureMockStore from 'redux-mock-store';

import mockState from '../../../../../../../../test/data/mock-state.json';
import { renderWithProvider } from '../../../../../../../../test/lib/render-helpers';
import {
  SignatureRequestSIWEWithResources,
  signatureRequestSIWE,
} from '../../../../../../../../test/data/confirmations/personal_sign';
import SIWESignInfo from './siwe-sign';

describe('SIWESignInfo', () => {
  it('renders correctly for SIWE signature request', () => {
    const state = {
      ...mockState,
      confirm: {
        currentConfirmation: signatureRequestSIWE,
      },
    };
    const mockStore = configureMockStore([])(state);
    const { container } = renderWithProvider(<SIWESignInfo />, mockStore);
    expect(container).toMatchSnapshot();
  });

  it('renders correctly for SIWE signature request with resources', () => {
    const state = {
      ...mockState,
      confirm: {
        currentConfirmation: SignatureRequestSIWEWithResources,
      },
    };
    const mockStore = configureMockStore([])(state);
    const { container } = renderWithProvider(<SIWESignInfo />, mockStore);
    expect(container).toMatchSnapshot();
  });
});
