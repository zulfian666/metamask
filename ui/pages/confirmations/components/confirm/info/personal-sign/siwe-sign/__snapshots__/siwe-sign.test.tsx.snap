// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SIWESignInfo renders correctly for SIWE signature request 1`] = `
<div>
  <div
    class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
    style="overflow-wrap: anywhere; min-height: 24px; position: relative;"
  >
    <div
      class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-flex-start"
      style="align-items: center;"
    >
      <p
        class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
      >
        Message
      </p>
    </div>
    <div
      class="mm-box mm-box--display-flex mm-box--gap-2 mm-box--flex-wrap-wrap mm-box--align-items-center mm-box--min-width-0"
    >
      <p
        class="mm-box mm-text mm-text--body-md mm-box--color-inherit"
        style="white-space: pre-wrap;"
      >
        I accept the MetaMask Terms of Service: https://community.metamask.io/tos
      </p>
    </div>
  </div>
  <div
    class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
    style="overflow-wrap: anywhere; min-height: 24px; position: relative;"
  >
    <div
      class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-flex-start"
      style="align-items: center;"
    >
      <p
        class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
      >
        URL
      </p>
    </div>
    <div
      class="mm-box mm-box--display-flex mm-box--gap-2 mm-box--flex-wrap-wrap mm-box--align-items-center mm-box--min-width-0"
    >
      <p
        class="mm-box mm-text mm-text--body-md mm-box--color-inherit"
        style="white-space: pre-wrap;"
      >
        metamask.github.io
      </p>
    </div>
  </div>
  <div
    class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
    style="overflow-wrap: anywhere; min-height: 24px; position: relative;"
  >
    <div
      class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-flex-start"
      style="align-items: center;"
    >
      <p
        class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
      >
        Network
      </p>
    </div>
    <div
      class="mm-box mm-box--display-flex mm-box--gap-2 mm-box--flex-wrap-wrap mm-box--align-items-center mm-box--min-width-0"
    >
      <p
        class="mm-box mm-text mm-text--body-md mm-box--color-inherit"
        style="white-space: pre-wrap;"
      >
        Ethereum Mainnet
      </p>
    </div>
  </div>
  <div
    class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
    style="overflow-wrap: anywhere; min-height: 24px; position: relative;"
  >
    <div
      class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-flex-start"
      style="align-items: center;"
    >
      <p
        class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
      >
        Account
      </p>
    </div>
    <div
      class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--align-items-center"
    >
      <div
        class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--align-items-center"
      >
        <div
          class="mm-box mm-text mm-avatar-base mm-avatar-base--size-xs mm-avatar-account mm-text--body-xs mm-text--text-transform-uppercase mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-default mm-box--background-color-background-alternative mm-box--rounded-full mm-box--border-color-transparent box--border-style-solid box--border-width-1"
        >
          <div
            class="mm-avatar-account__jazzicon"
          >
            <div
              style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 16px; height: 16px; display: inline-block; background: rgb(200, 20, 59);"
            >
              <svg
                height="16"
                width="16"
                x="0"
                y="0"
              >
                <rect
                  fill="#017B8E"
                  height="16"
                  transform="translate(0.39972303467835485 -0.4108005578484912) rotate(317.9 8 8)"
                  width="16"
                  x="0"
                  y="0"
                />
                <rect
                  fill="#F2B602"
                  height="16"
                  transform="translate(-3.7466980184561267 5.704967714142398) rotate(134.1 8 8)"
                  width="16"
                  x="0"
                  y="0"
                />
                <rect
                  fill="#FA8E00"
                  height="16"
                  transform="translate(-12.856136345377399 -5.7794405216344416) rotate(364.0 8 8)"
                  width="16"
                  x="0"
                  y="0"
                />
              </svg>
            </div>
          </div>
        </div>
        <p
          class="mm-box mm-text mm-text--body-md mm-box--margin-left-2 mm-box--color-inherit"
          data-testid="confirm-info-row-display-name"
        >
          0x935e7...05477
        </p>
      </div>
    </div>
  </div>
  <div
    class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
    style="overflow-wrap: anywhere; min-height: 24px; position: relative;"
  >
    <div
      class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-flex-start"
      style="align-items: center;"
    >
      <p
        class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
      >
        Version
      </p>
    </div>
    <div
      class="mm-box mm-box--display-flex mm-box--gap-2 mm-box--flex-wrap-wrap mm-box--align-items-center mm-box--min-width-0"
    >
      <p
        class="mm-box mm-text mm-text--body-md mm-box--color-inherit"
        style="white-space: pre-wrap;"
      >
        1
      </p>
    </div>
  </div>
  <div
    class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
    style="overflow-wrap: anywhere; min-height: 24px; position: relative;"
  >
    <div
      class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-flex-start"
      style="align-items: center;"
    >
      <p
        class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
      >
        Chain ID
      </p>
    </div>
    <div
      class="mm-box mm-box--display-flex mm-box--gap-2 mm-box--flex-wrap-wrap mm-box--align-items-center mm-box--min-width-0"
    >
      <p
        class="mm-box mm-text mm-text--body-md mm-box--color-inherit"
        style="white-space: pre-wrap;"
      >
        1
      </p>
    </div>
  </div>
  <div
    class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
    style="overflow-wrap: anywhere; min-height: 24px; position: relative;"
  >
    <div
      class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-flex-start"
      style="align-items: center;"
    >
      <p
        class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
      >
        Nonce
      </p>
    </div>
    <div
      class="mm-box mm-box--display-flex mm-box--gap-2 mm-box--flex-wrap-wrap mm-box--align-items-center mm-box--min-width-0"
    >
      <p
        class="mm-box mm-text mm-text--body-md mm-box--color-inherit"
        style="white-space: pre-wrap;"
      >
        32891757
      </p>
    </div>
  </div>
  <div
    class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
    style="overflow-wrap: anywhere; min-height: 24px; position: relative;"
  >
    <div
      class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-flex-start"
      style="align-items: center;"
    >
      <p
        class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
      >
        Issued
      </p>
    </div>
    <div
      class="mm-box mm-box--display-flex mm-box--gap-2 mm-box--flex-wrap-wrap mm-box--align-items-center"
    >
      <p
        class="mm-box mm-text mm-text--body-md mm-box--color-inherit"
        style="white-space: pre-wrap;"
      >
        30 September 2021, 16:25
      </p>
    </div>
  </div>
  <div
    class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
    style="overflow-wrap: anywhere; min-height: 24px; position: relative;"
  >
    <div
      class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-flex-start"
      style="align-items: center;"
    >
      <p
        class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
      >
        Request ID
      </p>
    </div>
    <div
      class="mm-box mm-box--display-flex mm-box--gap-2 mm-box--flex-wrap-wrap mm-box--align-items-center mm-box--min-width-0"
    >
      <p
        class="mm-box mm-text mm-text--body-md mm-box--color-inherit"
        style="white-space: pre-wrap;"
      >
        some_id
      </p>
    </div>
  </div>
</div>
`;

exports[`SIWESignInfo renders correctly for SIWE signature request with resources 1`] = `
<div>
  <div
    class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
    style="overflow-wrap: anywhere; min-height: 24px; position: relative;"
  >
    <div
      class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-flex-start"
      style="align-items: center;"
    >
      <p
        class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
      >
        Message
      </p>
    </div>
    <div
      class="mm-box mm-box--display-flex mm-box--gap-2 mm-box--flex-wrap-wrap mm-box--align-items-center mm-box--min-width-0"
    >
      <p
        class="mm-box mm-text mm-text--body-md mm-box--color-inherit"
        style="white-space: pre-wrap;"
      >
        I accept the MetaMask Terms of Service: https://community.metamask.io/tos
      </p>
    </div>
  </div>
  <div
    class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
    style="overflow-wrap: anywhere; min-height: 24px; position: relative;"
  >
    <div
      class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-flex-start"
      style="align-items: center;"
    >
      <p
        class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
      >
        URL
      </p>
    </div>
    <div
      class="mm-box mm-box--display-flex mm-box--gap-2 mm-box--flex-wrap-wrap mm-box--align-items-center mm-box--min-width-0"
    >
      <p
        class="mm-box mm-text mm-text--body-md mm-box--color-inherit"
        style="white-space: pre-wrap;"
      >
        metamask.github.io
      </p>
    </div>
  </div>
  <div
    class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
    style="overflow-wrap: anywhere; min-height: 24px; position: relative;"
  >
    <div
      class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-flex-start"
      style="align-items: center;"
    >
      <p
        class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
      >
        Network
      </p>
    </div>
    <div
      class="mm-box mm-box--display-flex mm-box--gap-2 mm-box--flex-wrap-wrap mm-box--align-items-center mm-box--min-width-0"
    >
      <p
        class="mm-box mm-text mm-text--body-md mm-box--color-inherit"
        style="white-space: pre-wrap;"
      >
        Ethereum Mainnet
      </p>
    </div>
  </div>
  <div
    class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
    style="overflow-wrap: anywhere; min-height: 24px; position: relative;"
  >
    <div
      class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-flex-start"
      style="align-items: center;"
    >
      <p
        class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
      >
        Account
      </p>
    </div>
    <div
      class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--align-items-center"
    >
      <div
        class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--align-items-center"
      >
        <div
          class="mm-box mm-text mm-avatar-base mm-avatar-base--size-xs mm-avatar-account mm-text--body-xs mm-text--text-transform-uppercase mm-box--display-flex mm-box--justify-content-center mm-box--align-items-center mm-box--color-text-default mm-box--background-color-background-alternative mm-box--rounded-full mm-box--border-color-transparent box--border-style-solid box--border-width-1"
        >
          <div
            class="mm-avatar-account__jazzicon"
          >
            <div
              style="border-radius: 50px; overflow: hidden; padding: 0px; margin: 0px; width: 16px; height: 16px; display: inline-block; background: rgb(200, 20, 59);"
            >
              <svg
                height="16"
                width="16"
                x="0"
                y="0"
              >
                <rect
                  fill="#017B8E"
                  height="16"
                  transform="translate(0.39972303467835485 -0.4108005578484912) rotate(317.9 8 8)"
                  width="16"
                  x="0"
                  y="0"
                />
                <rect
                  fill="#F2B602"
                  height="16"
                  transform="translate(-3.7466980184561267 5.704967714142398) rotate(134.1 8 8)"
                  width="16"
                  x="0"
                  y="0"
                />
                <rect
                  fill="#FA8E00"
                  height="16"
                  transform="translate(-12.856136345377399 -5.7794405216344416) rotate(364.0 8 8)"
                  width="16"
                  x="0"
                  y="0"
                />
              </svg>
            </div>
          </div>
        </div>
        <p
          class="mm-box mm-text mm-text--body-md mm-box--margin-left-2 mm-box--color-inherit"
          data-testid="confirm-info-row-display-name"
        >
          0x935E7...05477
        </p>
      </div>
    </div>
  </div>
  <div
    class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
    style="overflow-wrap: anywhere; min-height: 24px; position: relative;"
  >
    <div
      class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-flex-start"
      style="align-items: center;"
    >
      <p
        class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
      >
        Version
      </p>
    </div>
    <div
      class="mm-box mm-box--display-flex mm-box--gap-2 mm-box--flex-wrap-wrap mm-box--align-items-center mm-box--min-width-0"
    >
      <p
        class="mm-box mm-text mm-text--body-md mm-box--color-inherit"
        style="white-space: pre-wrap;"
      >
        1
      </p>
    </div>
  </div>
  <div
    class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
    style="overflow-wrap: anywhere; min-height: 24px; position: relative;"
  >
    <div
      class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-flex-start"
      style="align-items: center;"
    >
      <p
        class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
      >
        Chain ID
      </p>
    </div>
    <div
      class="mm-box mm-box--display-flex mm-box--gap-2 mm-box--flex-wrap-wrap mm-box--align-items-center mm-box--min-width-0"
    >
      <p
        class="mm-box mm-text mm-text--body-md mm-box--color-inherit"
        style="white-space: pre-wrap;"
      >
        1
      </p>
    </div>
  </div>
  <div
    class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
    style="overflow-wrap: anywhere; min-height: 24px; position: relative;"
  >
    <div
      class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-flex-start"
      style="align-items: center;"
    >
      <p
        class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
      >
        Nonce
      </p>
    </div>
    <div
      class="mm-box mm-box--display-flex mm-box--gap-2 mm-box--flex-wrap-wrap mm-box--align-items-center mm-box--min-width-0"
    >
      <p
        class="mm-box mm-text mm-text--body-md mm-box--color-inherit"
        style="white-space: pre-wrap;"
      >
        32891757
      </p>
    </div>
  </div>
  <div
    class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
    style="overflow-wrap: anywhere; min-height: 24px; position: relative;"
  >
    <div
      class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-flex-start"
      style="align-items: center;"
    >
      <p
        class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
      >
        Issued
      </p>
    </div>
    <div
      class="mm-box mm-box--display-flex mm-box--gap-2 mm-box--flex-wrap-wrap mm-box--align-items-center"
    >
      <p
        class="mm-box mm-text mm-text--body-md mm-box--color-inherit"
        style="white-space: pre-wrap;"
      >
        30 September 2021, 16:25
      </p>
    </div>
  </div>
  <div
    class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
    style="overflow-wrap: anywhere; min-height: 24px; position: relative;"
  >
    <div
      class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-flex-start"
      style="align-items: center;"
    >
      <p
        class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
      >
        Request ID
      </p>
    </div>
    <div
      class="mm-box mm-box--display-flex mm-box--gap-2 mm-box--flex-wrap-wrap mm-box--align-items-center mm-box--min-width-0"
    >
      <p
        class="mm-box mm-text mm-text--body-md mm-box--color-inherit"
        style="white-space: pre-wrap;"
      >
        some_id
      </p>
    </div>
  </div>
  <div
    class="mm-box confirm-info-row mm-box--margin-top-2 mm-box--margin-bottom-2 mm-box--padding-right-2 mm-box--padding-left-2 mm-box--display-flex mm-box--flex-direction-row mm-box--flex-wrap-wrap mm-box--justify-content-space-between mm-box--color-text-default mm-box--rounded-lg"
    style="overflow-wrap: anywhere; min-height: 24px; position: relative;"
  >
    <div
      class="mm-box mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-center mm-box--align-items-flex-start"
      style="align-items: center;"
    >
      <p
        class="mm-box mm-text mm-text--body-md-medium mm-box--color-inherit"
      >
        Resources
      </p>
    </div>
    <div
      class="mm-box mm-box--display-flex mm-box--gap-2 mm-box--flex-wrap-wrap mm-box--align-items-center mm-box--min-width-0"
    >
      <p
        class="mm-box mm-text mm-text--body-md mm-box--color-inherit"
        style="white-space: pre-wrap;"
      >
        ipfs://Qme7ss3ARVgxv6rXqVPiikMJ8u2NLgmgszg13pYrDKEoiu
      </p>
    </div>
    <div
      class="mm-box mm-box--display-flex mm-box--gap-2 mm-box--flex-wrap-wrap mm-box--align-items-center mm-box--min-width-0"
    >
      <p
        class="mm-box mm-text mm-text--body-md mm-box--color-inherit"
        style="white-space: pre-wrap;"
      >
        https://example.com/my-web2-claim.json
      </p>
    </div>
  </div>
</div>
`;
